src/app/public/modules/autocomplete/autocomplete.component.ts

```diff
@@ -313,33 +364,25 @@ export class SkyAutocompleteComponent
       ].join(' '));
     }
 
-    this.inputDirective.displayWith = this.descriptorProperty;
-
-    this.inputDirective.textChanges
-      .pipe(
-        takeUntil(this.ngUnsubscribe),
-        debounceTime(this.debounceTime)
-      )
-      .subscribe((change) => {
-        this.searchTextChanged(change.value);
-      });
-
-    this.inputDirective.blur
-      .pipe(
-        takeUntil(this.ngUnsubscribe)
-      )
-      .subscribe(() => {
-        this.searchText = '';
-        this.closeDropdown();
-      });
-
+    if (this.themeSvc) {
+      this.themeSvc.settingsChange
+        .pipe(
+          takeUntil(this.ngUnsubscribe)
+        )
+        .subscribe((themeSettings) => {
+          this.themeName = themeSettings.currentSettings?.theme?.name;
+        });
+    }
   }
 
   public ngAfterViewInit(): void {
     this.addInputEventListeners();
   }
 
   public ngOnDestroy(): void {
+    this.inputDirectiveUnsubscribe.next();
+    this.inputDirectiveUnsubscribe.complete();
+    this.inputDirectiveUnsubscribe = undefined;
```

Comment:
Heads up - this is a pattern I think we should get away from... We've had a few reports about consumers' logs recording errors caused by having an undefined observable/subject. Somehow, the observable is getting set to `undefined` before it has a chance to call methods on it, resulting in this error: `You provided 'undefined' where a stream was expected.` See [this bug](https://github.com/blackbaud/skyux-core/issues/185) for more info.

How would you feel about removing the `undefined` line?

Reply:
I'm fine with removing it. In all honesty, I don't normally add that line but I did here because we were already doing it for the `ngUnsubscribe` in this component and it felt wrong to be inconsistent in one component. I've removed both of the lines that set them to `undefined`.